api/src/main/java/org/eclipse/microprofile/lra/annotation/Compensate.java

```diff
@@ -27,42 +27,131 @@
  * the implementation of this specification MUST prohibit successful startup of the application
  * (e.g. with a runtime exception).
+ * </p>
+ *
+ * <p>
+ * If the participant cannot compensate immediately then it must report that the
+ * compensation is in progress by either returning a future (such as
+ * {@link java.util.concurrent.CompletionStage}) which will eventually report
+ * one of the final states, or a <code>202 Accepted</code> JAX-RS response code or,
+ * in the case of non JAX-RS resource methods, by returning
+ * {@link ParticipantStatus#Compensating} (see the specification
+ * document for more details).
+ * </p>
  *
- * Note that, according to the state model {@link LRAStatus} once an LRA has been
- * asked to cancel it is no longer possible to join with it as a participant.
- * Therefore in JAX-RS, combining this annotation with an `@LRA` annotation that does not
- * start a new LRA will result in a `412 PreCondition Failed` status code and is
- * not advised. On the other hand, combining it with an `@LRA` annotation that
- * begins a new LRA can in certain use case make sense, but in this case the LRA
+ * <p>
+ * Note that, according to the state model defined by {@link LRAStatus}, it is not possible
+ * to receive compensation notifications after an LRA has been asked to cancel.
+ * Therefore combining this annotation with an <code>&#64;LRA</code> annotation that does not
+ * start a new LRA will result in a <code>412 PreCondition Failed</code> JAX-RS response
+ * code. On the other hand, combining it with an <code>&#64;LRA</code> annotation that
+ * begins a new LRA can in certain uses case make sense, but in this case the LRA
  * that this method is being asked to compensate for will be unavailable.
+ * </p>
+ *
+ * <p>
+ * If the method is a JAX-RS resource method (or is a non JAX-RS method
+ * annotated with <code>&#64;Compensate</code> with return type
+ * <code>javax.ws.rs.core.Response)</code> then the following are the only
+ * valid response codes:
+ * </p>
+ *
+ *<table border="0" cellpadding="3" cellspacing="0"
+ *   summary="Valid JAX-RS compensation response codes">
+ * <caption>
+ *     <span>JAX-RS Compensation Response Codes</span>
+ *     <span>&nbsp;</span>
+ * </caption>
+ * <tr>
+ *   <th scope="col">Code</th>
+ *   <th scope="col">Response Body</th>
+ *   <th scope="col">Meaning</th>
+ * </tr>
+ * <tr>
+ *   <td scope="row">200</td>
+ *   <td scope="row">Empty</td>
+ *   <td scope="row">The resource has successfully compensated</td>
+ * </tr>
+ * <tr>
+ *   <td scope="row">202</td>
+ *   <td scope="row">Empty</td>
+ *   <td scope="row">The resource is still attempting compensation</td>
+ * </tr>
+ * <tr>
+ *   <td scope="row">404</td>
+ *   <td scope="row">Empty</td>
+ *   <td scope="row">The resource does not know about the LRA</td>
+ * </tr>
+ * <tr>
+ *   <td scope="row">412</td>
+ *   <td scope="row">Empty</td>
+ *   <td scope="row">The resource cannot join the LRA because the LRA is no
```

Comment:
This is Compensate annotation. We shouldn't be mentioning joining here I think.

Reply:
Good catch. I removed it from the Complete javadoc too.

Reply:
Resolving here but do note that the text is added back due to a latter comment from the reviewer.